functions/Get-DbaLastBackup.ps1

```diff
@@ -0,0 +1,158 @@
+﻿Function Get-DbaLastBackup
+{
+<#
+.SYNOPSIS
+Get date/time for last known backups
+
+.DESCRIPTION
+Retrieves and compares the date/time for the last known backups, as well as the creation date/time for the database.
+
+
+.PARAMETER SqlServer
+The SQL Server that you're connecting to.
+
+.PARAMETER Credential
+Credential object used to connect to the SQL Server as a different user
+
+.PARAMETER Databases
+Return information for only specific databases
+
+.PARAMETER Exclude
+Return information for all but these specific databases
+
+.PARAMETER Detailed
+Shows detailed information
+
+.NOTES
+Copyright (C) 2016 Klaas Vandenberghe (@powerdbaklaas)
+
+This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+.LINK
+
+
+.EXAMPLE
+Get-DbaLastBackup -SqlServer ServerA\sql987
+
+Returns a custom object with Server name, Database name, and the date the last time backups were performed
+
+.EXAMPLE
+Get-DbaLastBackup -SqlServer ServerA\sql987 -Detailed | Out-Gridview
+
+Returns a gridview displaying Server, Database, LastFullBU, TimeSinceFullBU, LastDiffBU, TimeSinceDiffBU, LastLogBU, TimeSinceLastLogBU, RecoveryModel, Status, DatabaseCreated, DaysSinceDbCreated
+
+#>
+	[CmdletBinding()]
+	Param (
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Alias("ServerInstance", "SqlInstance")]
+		[string[]]$SqlServer,
+		[PsCredential]$Credential,
+		[switch]$Detailed
+	)
+
+	DynamicParam {
+		if ($SqlServer) {
+			return Get-ParamSqlDatabases -SqlServer $SqlServer[0] -SqlCredential $Credential
+		}
+	}
+
+	BEGIN
+	{
+		# Convert from RuntimeDefinedParameter object to regular array
+		$databases = $psboundparameters.Databases
+		$exclude = $psboundparameters.Exclude
+	}
+
+	PROCESS
+	{
+		foreach ($servername in $SqlServer)
+		{
+			try
+			{
+				$server = Connect-SqlServer -SqlServer $servername -SqlCredential $Credential

```

Comment:
This is the kind of error which caused the command to stop processing, it was an instance still present in my Central Management Server which is no longer running at all, so it probably isn't fair to say you're not handling failed connections, just this specific kind of failure... ?

There has been talk about improving Connect-SqlServer to handle this kind of thing centrally, if I recall correctly.
-------

Can't connect to blah\meh: System.Data.SqlClient.SqlException (0x80131904): A network-related or instance-specific error occurred while establishing a connection 
to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote 
connections. (provider: SQL Network Interfaces, error: 26 - Error Locating Server/Instance Specified)
    
At C:\Users\DanielAlexander\My Documents\WindowsPowerShell\Modules\dbatools\internal\Connect-SqlServer.ps1:109 char:3
+         throw "Can't connect to $sqlserver`: $message "
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Can't connect t...pecified)
    :String) [], RuntimeException
    + FullyQualifiedErrorId : Can't connect to blah\meh: System.Data.SqlClient.SqlException (0x80131904): A network-related or instance-specific error occurred w 
   hile establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is  
   configured to allow remote connections. (provider: SQL Network Interfaces, error: 26 - Error Locating Server/Instance Specified)